lib/spack/llnl/util/lock.py

```diff
@@ -512,7 +594,7 @@ def downgrade_write_to_read(self, timeout=None):
         """
         timeout = timeout or self.default_timeout
 
-        if self._writes == 1 and self._reads == 0:
+        if self._writes == 1:
```

Comment:
It seems good to remove this check, but if we are multiply-nested inside of read locks, I'm worried about a few lines below this where we set `self._reads= 1`. My first thought is that we should actually do `self._reads += 1` (which I think is correct because `acquire_write()` doesn't increment `self._reads`.